### ISSUE TITLE
Tutorial or docs on human skin/hair shaders

### ISSUE THREAD

**Issue by Zireael07** [godotengine/godot-docs#7398]
**Your Godot version:**
4.0.2

**Issue description:**
It's supposedly possible to achieve proper skin/hair look with built-in materials and shaders. The problem is, next to no one knows how to do it. Please document.

Raised in a comment in https://github.com/godotengine/godot-proposals/issues/6900 which was closed because OP deleted comments/other assets were found, so I'm opening an issue solely for the **documentation part**

**URL to the documentation page (if already existing):**
N/A

**Comment by skyace65** [comment]
I'm going to say that a tutorial like this should not be added to the docs, we moved away from tutorial style documentation towards manual style documentation.

**Comment by mhilbrunner** [comment]
Yeah, I'm tending to agree this is out of scope with our current resources. This would be a good fit for a community plugin/tutorial/demo with docs.

**Comment by Zireael07** [comment]
I said "tutorial OR docs", so I see no issue with manual-style explanation how to achieve the effect

**Comment by mhilbrunner** [comment]
The difference is larger than just the naming: the idea with the manual is that we document and explain the engine features - in this case, how PBR and shaders and lighting works. Making use of this functionality to build your specific game is up to you (or community plugins, assets and tutorials).

This change arose due to us previously trying to create such tutorials and those just end up unmaintained and outdated because we lack the resources to maintain them properly.

Skin/hair kinda straddles the line where it's somewhat specific and a large topic but at the same time its useful for many games. I guess if someone in the community steps up to make a good tutorial and that is maintained for some time we can see about making it official.

**Comment by and-rad** [comment]
Seeing as how specific this topic is, I think the demo projects repo might be better suited for it.

**Comment by GeorgeS2019** [comment]
An **alternative** option to this [discussion](https://github.com/barcoderdev/unitypackage_godot/issues/13#issue-1721410489)

**Comment by GeorgeS2019** [comment]
We need to expand the topic on **Godot Digital Human** in all aspects as demonstrated by Nvidia Omniverse
https://github.com/RoboPoets/digital_human
![image](https://github.com/godotengine/godot-docs/assets/49812372/f30d2a5a-4e3e-48e0-8194-25c001117a01)
https://www.youtube.com/watch?v=ss13J1emsvw
![image](https://github.com/godotengine/godot-docs/assets/49812372/089899ac-62e8-43d7-ae02-d3b12cb2bb27)

**Comment by GeorgeS2019** [comment]
https://www.youtube.com/watch?v=ft--lVFpx4E&pp=ygUUZ29kb3Q0IGRpZ2l0YWwgaHVtYW4%3D

**Comment by LiveTrower** [comment]
I adapted the Unreal Engine Marschner BSDF and Kajiya-Kay BSDF for Godot to enhance hair shading, achieving more realistic and lifelike representations of human hair. https://github.com/LiveTrower/Godot-Hair-Shading

Additionally, I created a demo using free models from [Character Creator 4](https://www.reallusion.com/character-creator/download.html), incorporating my hair shader and eye shaders from [digital_human](https://github.com/RoboPoets/digital_human). However, I haven’t made it public yet as I’m unsure of the appropriate licensing for its release.

![Character1](https://github.com/user-attachments/assets/426a749f-ad79-4d66-9af8-54a1b96bc19b)
![Character2](https://github.com/user-attachments/assets/ac28a2b5-377e-4ca8-8f7e-6e09a7cf40ac)

**Comment by renight0** [comment]
Hey @LiveTrower . Can we already try using these shaders in our projects?

 I don't understand what is the problem with the licensing. Aren't they both MIT licensed?

**Comment by LiveTrower** [comment]
@renight0 There's no problem you can use it; in fact, I made it so that all people can use it. Regarding the licensing issue, I was referring to the free models from Character Creator 4, not the shaders themselves.

**Comment by renight0** [comment]
@LiveTrower @and-rad Could you add how to import these shaders into the readMe file? I'm relatively new to godot and I'm a 3D artist but I tried bringing the realistics skin and eye shader to godot and I'm getting broken dependencies errors when I try oppening the scenes 

![Image](https://github.com/user-attachments/assets/e2a78d31-e0c1-4022-aa27-4c1283190d1e)

**Comment by and-rad** [comment]
In the case of the Digital Human addon, you import the shaders into a project by importing the entire addon. That's done the same way as importing any other addon. Maybe that's the missing step?

**Comment by renight0** [comment]
> In the case of the [Digital Human](https://github.com/RoboPoets/digital_human) addon, you import the shaders into a project by importing the entire addon. That's done the same way as importing any other addon. Maybe that's the missing step?

The right way to import addons is by creating a folder called "addons", copying the addon folder (digital human) content to the addon folder and enabling the plugin in the project setting>plugin, right? Just like in this video:

https://www.youtube.com/watch?v=b6Zh9Wpq8lc

and according to the documentation: https://docs.godotengine.org/en/stable/tutorials/plugins/editor/installing_plugins.html

So I created a new project just to import the addon the right way and see if it worked. And I got these errors

![Image](https://github.com/user-attachments/assets/300591cd-43e9-4367-8e0a-9a5add088b87)

![Image](https://github.com/user-attachments/assets/30988535-9798-451c-b0ff-c8f9985553ac)

Idk if I'm doing something wrong when importing :(

**Comment by and-rad** [comment]
The addon's folder name is `digital_human`, not `digital_human-main`. You probably downloaded the addon as a zip file straight from GitHub instead of cloning it. Rename the folder and the errors should go away.

**Comment by renight0** [comment]
@and-rad Thanks a lot! I think the shaders are now working properly.

![Image](https://github.com/user-attachments/assets/6aff2724-f04b-43df-a33f-e179f8fd0d4b)

I wonder how it would be possible to add a parallax occlusion mapping feature to the eye shader like J Hill mentioned in this video:

https://youtu.be/bbJmtVCdOh0?si=bRaurBiSR135vu63&t=328

**Comment by and-rad** [comment]
The eye shader has parallax occlusion mapping in the `eye_textured.tres` variation of the shader. The settings are in the `Displacement` section, the POM texture itself can be found in the `Textures` section of the shader.

**Comment by renight0** [comment]
@and-rad Thank you!!

**Comment by renight0** [comment]
@and-rad by the way, I really wanted to study the eye mesh model and UVs from your addon. I can't find the model in the addon folder. Could you provide a link to it?

**Comment by and-rad** [comment]
The model is part of the `showcase.glb` file. Just import that one into Blender and the entire map is there for you to edit and inspect. Since GLTF converts all quads to triangles, the first thing I'd do is `Alt+J` on the imported mesh in edit mode, makes it easier to make sense of the topology.
